### PR TITLE
Parse xml for font and layout

### PR DESCRIPTION
**Pull request by t-bullock**
I'll try to create a new branch for each pull request in the future.

### CODE REVIEW

File: pdfmaker/kassia.py [hunk 1]
```diff
@@ -77,18 +79,24 @@ def createPDF(self):
...
             c.setFillColor(colors.black)
 
-            neumesText = " ".join(troparion.find('neumes').text.strip().split())
-            lyricsText = " ".join(troparion.find('lyrics').text.strip().split())
+            # Get attributes for neumes
+            neume_elem = troparion.find('neumes')
+            if (neume_elem is not None):
+                neumesText = " ".join(neume_elem.text.strip().split())
+                neume_attrib = neume_elem.attrib
+                neume_attrib = self.fill_title_dict(neume_attrib)
```
Comment (jdaly101) [inline pdfmaker/kassia.py]: Doesn't have to be in this PR, but we should toss up a warning if the element is non-existent. I could see someone not wanting lyrics (blank templates for prosomia), but neumes seem like a "must have" to me. Thoughts?

File: pdfmaker/kassia.py [hunk 2]
```diff
@@ -43,17 +43,19 @@ def parseFile(self):
...
-        self.nFontSize = 20
-        self.lFontSize = 12
-        
+        self.topMargin = margin_attrib['top_margin']
+        self.bottomMargin = margin_attrib['bottom_margin']
+        self.leftMargin = margin_attrib['left_margin']
+        self.rightMargin = margin_attrib['right_margin']
+        self.lineHeight = margin_attrib['line_height']
+        self.lineWidth = self.paperSize[0] - (self.leftMargin + self.rightMargin)
+
```
Comment (jdaly101) [inline pdfmaker/kassia.py]: I like the abstraction, but these need reasonable defaults if they're left unspecified.

File: pdfmaker/kassia.py [hunk 3]
```diff
@@ -239,6 +247,14 @@ def linebreak(self,neumes,lyrics = None):
...
         return neumePos, lyricPos
 
+    def fill_page_dict(self, page_dict):
+        for attrib_name in page_dict:
+            if not page_dict.has_key(attrib_name):
+                page_dict[attrib_name] = 72
+            else:
+                page_dict[attrib_name] = int(page_dict[attrib_name])
+        return page_dict
+
```
Comment (jdaly101) [inline pdfmaker/kassia.py:280]: Oh I see, you're stuffing defaults in here. Hmmm...

File: pdfmaker/kassia.py (hunk 1 above)
Comment (jdaly101) [inline pdfmaker/kassia.py]: Also, isn't this setting the default font for neumes as EZOmega?

Comment (t-bullock) [inline pdfmaker/kassia.py]: Yeah, we should definitely allow for no lyrics. I'll add in another pull request.

I added a separate dictionary filling function for lyrics and neumes so they aren't forced into any defaults from the title.

File: pdfmaker/kassia.py (hunk 3 above)
Comment (t-bullock) [inline pdfmaker/kassia.py:280]: Yeah this wasn't the best (just quick and easy). I moved the default settings back up towards the top of createPDF().

**Comment by jdaly101** [comment]
Looks good! :ship:

**Pull request merged**